pkg/reconciler/common/stages.go

```diff
@@ -110,6 +111,11 @@ func DeleteObsoleteResources(ctx context.Context, instance v1alpha1.KComponent,
 	installed, err := fetch(ctx, instance)
 	if err != nil {
 		logger.Error("Unable to obtain the installed manifest; obsolete resources may linger", err)
+		// Report an error in `Status`, issue ref: https://github.com/knative/operator/issues/741
```

Comment:
I do not think this is the best place to add the message in the status.
Since we set spec.version which is not supported by the current version, we need to raise the error, where we check the spec.version and get the target manifests.
And report "the target version is too old, unable to get the target manifests or something".

Reply:
Yeah I agree with you, that's exactly why I ask whether we should reject in the issue.

Reply:
https://github.com/knative/operator/pull/750  I opened another PR. I think that's a better solution to resolve this issue.
I users specify spec.manifests with the version, operator will allow that.